solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/outbound/JCSMPOutboundMessageHandler.java

```diff
@@ -233,6 +235,18 @@ public void start() {
 		}
 
 		try {
+			Map<String, String> headerNameMapping = properties.getExtension().getHeaderNameMapping();
+			if (headerNameMapping != null && !headerNameMapping.isEmpty()) {
+				Set<String> uniqueTargetHeaderNames = new HashSet<>(headerNameMapping.values());
+				if (uniqueTargetHeaderNames.size() < headerNameMapping.size()) {
+					MessagingException exception = new MessagingException(String.format(
+							"Two or more headers map to the same header name in headerNameMapping %s <outbound adapter %s>",
+							properties.getExtension().getHeaderNameMapping(), id));
+					LOGGER.warn(exception.getMessage());
+					throw exception;
+				}
+			}
```

Comment:
given that this rule needs to be checked on `start()` for so many components (async consumer, polled consumer, producer), maybe we should create a custom constraint for this to redistribute and validate the the property? https://docs.spring.io/spring-framework/reference/core/validation/beanvalidation.html#validation-beanvalidation-spring-constraints